_source/_posts/2022-01-05-eleventy.md

```diff
@@ -0,0 +1,375 @@
+
+```HTML
+---
+layout: _layout_unauthenticated.11ty.js
+---
+
+<div class="row align-items-md-stretch">
+    <div class="col-md-12">
+        <div class="h-100 p-5 text-white bg-secondary rounded-3">
+            <h2>Using Okta with 11ty</h2>
+            <p>A small tutorial on writing static sites with 
+                <a class="text-dark" href="https://www.11ty.dev/" target="_blank">11ty</a>, 
+                secured by <a class="text-dark" href="https://www.okta.com/" target="_blank">Okta</a>
+                written by <a class="text-dark" href="https://profile.fishbowlllc.com/" target="_blank">Nik Fisher.</a>
+            </p>
+            <a class="btn btn-outline-light" role="button" href="Login">Login Now</a>
+            <a class="btn btn-light" role="button" href="README">README</a>
+        </div>
+    </div>
+</div>
+```
+
+You will notice the `layout` variable at the top of the.  You will add your layout pages later in this tutorial but take notes that you will need to assign a layout to each page if you wish.
+
+Next, create a file for `login.html` and add the following code.  
+
+```HTML
+---
+layout: _layout_unauthenticated.11ty.js
+---
+
+<div id="okta-login-container"></div>
+<script>
+  oktaSignIn.authClient.token.getUserInfo().then(function(user) {
+    window.location='/protected';
+  },function(error) {
+    oktaSignIn.showSignInToGetTokens({
+      el: '#okta-login-container'
+    }).then(function(tokens) {
+      oktaSignIn.authClient.tokenManager.setTokens(tokens);
+      oktaSignIn.remove();
+      window.location='/protected';
+    }).catch(function(err) {
+      console.error(err);
+    });
+  });
+</script>
+```
+
+This page has a bit more to it.  The only HTML here is a `div` with `id="okta-login-container"`.  You call the `OktaSignIn` features and display the login screen to the user in The javascript of this page.  Okta will take care of creating the login form and handling the actual authentication.  All you need to do is capture the token on the back and store the token.  Once you have done that you will redirect the user to the `protected` page.  
+
+You can add the protected page now.  Add the code below to it.
+
+```HTML
+---
+layout: _layout_authenticated.11ty.js
+---
+
+<div id="content"></div>
+
+<script>
+    function loadPage(){
+        //replace with API call 
+        document.getElementById('content').innerHTML = "<h2>Congrats</h2><h5>You've reached a protected page</h5>"
+    }
+</script>
+```
+
+Since you haven't seen the `_layout_authenticated.11ty.js` page yet, this will need some explanation.  The `loadPage` function is called the layout after verifying the user is authenticated.  There's a comment to replace the code here with an API call.  YOu could accomplish this with an API or a serverless implementation using something like AWS Lambda.  In either case, you could pass the token in the token and verify it on the server side before displaying any content to the user.
```

Comment:
```suggestion
Since you haven't seen the `_layout_authenticated.11ty.js` page yet, this will need some explanation.  The `loadPage` function is called the layout after verifying the user is authenticated.  There's a comment to replace the code here with an API call.  You could accomplish this with an API or a serverless implementation using something like AWS Lambda.  In either case, you could pass the token in the token and verify it on the server side before displaying any content to the user.
```